### PR TITLE
add envvars.log to the flare

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Add an `envvars.log` file in the flare, that is a text dump of `os.Environ()`.

### Motivation

Although we already have `runtime_config_dump.yaml` that will allow us to detect envvar bindings, the docker library relies on [other envvars](https://docs.docker.com/engine/reference/commandline/cli/#environment-variables), that viper is not aware of.

### CODE REVIEW

**Comment by masci** [comment]
Are we comfortable relying on our set of replacers? The risk of leaking sensitive info is high here, would it be possible/useful to whitelist well known env var names like `DOCKER_*`?

**Comment by xvello** [comment]
Although very low probability a rogue `GOGC`, `GODEBUG`, or `GOMAXPROCS` might have impact over the agent's behaviour. That could be useful too.

I'm good with either going behind the `isContainerized` condition or adding a whitelist mechanism to get these variables if they are set.

**Comment by masci** [comment]
I'd go with the `isContainerized` guard: it's fast, works for you and we can discuss the whitelisting approach later, when needed.

File: pkg/flare/archive.go [hunk 1]
```diff
@@ -212,6 +218,28 @@ func zipDiagnose(zipFile *archivex.ZipFile, hostname string) error {
 	return nil
 }
 
+func zipEnvvars(zipFile *archivex.ZipFile, hostname string) error {
+	var b bytes.Buffer
+
+	for _, envvar := range os.Environ() {
+
```
Comment (masci) [inline pkg/flare/archive.go]: Can we remove this blank line? Sorry it's my OCD speaking 🙈

**Comment by hkaj** [comment]
Agreed, limiting the env var collection to setups where we can access docker from inside the container is not ideal. However if we run in a container and without docker access, chances are we run as a side car, and have access to sensitive env vars.

`IsContainerized` is a weak barrier, it just checks for `DOCKER_DD_AGENT == yes`. If the agent is in the same container as an app that takes creds as env vars we're in trouble. If a customer hacks its config to leverage some container-related feature and we dump all env vars from their host that's not great either.

If we're looking for some specific env vars we should list them and start with a whitelist directly. As is it seems quite a risk to me.

File: pkg/flare/envvars.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"GOMAXPROCS=",
+	"GOTRACEBACK=",
+}
+
+func getWhitelistedEnvvars() []string {
+	var vars []string
+	for _, envvar := range os.Environ() {
+		for _, prefix := range envvarPrefixWhitelist {
+			if strings.HasPrefix(envvar, prefix) {
+				vars = append(vars, envvar)
```
[the remark this answers is not in the document]
Comment (xvello) [reply pkg/flare/envvars.go]: indeed, added

**Review by hkaj — APPROVED** [review]
💯

**Pull request merged**